app/ethos/views.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+'''
+    Copyright (C) 2017 Gitcoin Core
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+'''
+
+import json
+
+from django.conf import settings
+from django.http import Http404, JsonResponse
+from django.template.response import TemplateResponse
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+from django.views.decorators.csrf import csrf_exempt
+
+import twitter
+from ethos.models import Hop, ShortCode
+from gas.utils import recommend_min_gas_price_to_confirm_in_time
+from ratelimit.decorators import ratelimit
+from retail.helpers import get_ip
+from web3 import HTTPProvider, Web3
+
+w3 = Web3(HTTPProvider(settings.WEB3_HTTP_PROVIDER))
+
+
+@csrf_exempt
+@ratelimit(key='ip', rate='5/m', method=ratelimit.UNSAFE, block=True)
+def redeem_coin(request, shortcode):
+    print(shortcode)
+
+    if request.body:
+        status = 'OK'
+
+        body_unicode = request.body.decode('utf-8')
+        body = json.loads(body_unicode)
+
+        try:
+            twitter_api = get_twitter_api()
+
+            if not twitter_api:
+                raise Exception(_('Error while setting up Twitter API'))
```

Comment:
Can we handle these `Exception` entries by providing the user with the associated message versus raising an exception?

Reply:
My main concern here is that we're wrapping a good majority of the view  to catch on any exception.  If any exception occurs, whether it's your intended `raise Exception` or not, it's going to fall through as if it's intended handling for the generic exception.  We could probably define our own exception in this case and/or make use of `twitter.errors.TwitterError` throughout.